include/boost/geometry/algorithms/azimuth.hpp

```diff
@@ -0,0 +1,226 @@
+            >::type calc_t;
+
+        calc_t result = 0;
+        calc_t x1 = geometry::get_as_radian<0>(p1);
+        calc_t y1 = geometry::get_as_radian<1>(p1);
+        calc_t x2 = geometry::get_as_radian<0>(p2);
+        calc_t y2 = geometry::get_as_radian<1>(p2);
+
+        strategy.azimuth().apply(x1, y1, x2, y2, result);
+
+        // NOTE: It is not clear which units we should use for the result.
+        //   For now radians are always returned but a user could expect
+        //   e.g. something like this:
+        /*
+        bool const both_degree = std::is_same
+                <
+                    typename detail::cs_angular_units<Point1>::type,
+                    geometry::degree
+                >::value
+            && std::is_same
+                <
+                    typename detail::cs_angular_units<Point2>::type,
+                    geometry::degree
+                >::value;
+        if (both_degree)
+        {
+            result *= math::r2d<calc_t>();
+        }
+        */
+
+        return result;
+    }
+};
+
+} // namespace dispatch
+#endif // DOXYGEN_NO_DISPATCH
+
+
+namespace resolve_strategy
+{
+
+template
+<
+    typename Strategy,
+    bool IsUmbrella = strategies::detail::is_umbrella_strategy<Strategy>::value
+>
+struct azimuth
+{
+    template <typename P1, typename P2>
+    static auto apply(P1 const& p1, P2 const& p2, Strategy const& strategy)
+    {
+        return dispatch::azimuth<P1, P2>::apply(p1, p2, strategy);
+    }
+};
+
+template <typename Strategy>
+struct azimuth<Strategy, false>
+{
+    template <typename P1, typename P2>
+    static auto apply(P1 const& p1, P2 const& p2, Strategy const& strategy)
+    {
+        using strategies::azimuth::services::strategy_converter;
+        return dispatch::azimuth
+            <
+                P1, P2
+            >::apply(p1, p2, strategy_converter<Strategy>::get(strategy));
+    }
+};
+
+template <>
+struct azimuth<default_strategy, false>
+{
+    template <typename P1, typename P2>
+    static auto apply(P1 const& p1, P2 const& p2, default_strategy)
+    {
+        typedef typename strategies::azimuth::services::default_strategy
+            <
+                P1, P2
+            >::type strategy_type;
+
+        return dispatch::azimuth<P1, P2>::apply(p1, p2, strategy_type());
+    }
+};
+
+
+} // namespace resolve_strategy
+
+
+namespace resolve_variant
+{
+} // namespace resolve_variant
+
+
+/*!
+\brief Calculate azimuth of a segment defined by a pair of points.
+\ingroup azimuth
+\tparam Point1 Type of the first point of a segment.
+\tparam Point2 Type of the second point of a segment.
+\param point1 First point of a segment.
+\param point2 Second point of a segment.
+\return Azimuth in radians.
+
+\qbk{[include reference/algorithms/azimuth.qbk]}
+
+\qbk{
+[heading Example]
+[azimuth]
+[azimuth_output]
+}
+*/
+template <typename Point1, typename Point2>
+inline auto azimuth(Point1 const& point1, Point2 const& point2)
```

Comment:
Cool, this `auto` here!
:smiley:

Reply:
Yes, the same could be done in `distance()` and `area()`!